homeassistant/components/sensor/isy994.py

```diff
@@ -309,3 +312,47 @@ def unit_of_measurement(self) -> str:
             return self.hass.config.units.temperature_unit
         else:
             return raw_units
+
+
+class ISYWeatherDevice(isy.ISYDevice):
+    """Representation of an ISY994 weather device."""
+
+    _domain = 'sensor'
+
+    def __init__(self, node) -> None:
+        """Initialize the ISY994 weather device."""
+        isy.ISYDevice.__init__(self, node)
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique identifier for the node."""
+        return self._node.name
+
+    @property
+    def raw_units(self) -> str:
+        """Return the raw unit of measurement."""
+        if self._node.uom == 'F':
+            return TEMP_FAHRENHEIT
+        if self._node.uom == 'C':
+            return TEMP_CELSIUS
+        return self._node.uom
+
+    @property
+    def state(self) -> object:
+        """Return the value of the node."""
+        # pylint: disable=protected-access
+        val = self._node.status._val
+        raw_units = self._node.uom
```

Comment:
Not this time. The weather data in the ISY library isn't stored in the same Node class. In order to make the ISYDevice abstract class to mostly work with weather data, I create a named tuple with the data. In that named tuple, UOM is just a string.